acpi/src/madt.rs

```diff
@@ -648,3 +728,7 @@ fn parse_mps_inti_flags(flags: u16) -> crate::AcpiResult<(Polarity, TriggerMode)
 
     Ok((polarity, trigger_mode))
 }
+
+fn timeout_to_loops(timeout: Duration) -> u64 {
```

Comment:
Does this just assume each loop iteration will take around a microsecond to complete? If we're not actually timing it, I think it would be better just to ask the user for the raw number of iterations to try?